Library/Homebrew/livecheck/livecheck.rb

```diff
@@ -559,6 +582,9 @@ def latest_version(
       livecheck_regex = livecheck.regex || referenced_livecheck&.regex
       livecheck_strategy = livecheck.strategy || referenced_livecheck&.strategy
       livecheck_strategy_block = livecheck.strategy_block || referenced_livecheck&.strategy_block
+      homebrew_curl = if [:page_match, :header_match].include?(livecheck_strategy)
+        homebrew_curl?(formula_or_cask, livecheck_url)
+      end
```

Comment:
`homebrew_curl` is true or false only if either above strategies are being used, and nil otherwise. This allows the `homebrew_curl` argument to fall under the `_unused: T.nilable` parameter when other strategies are active, avoiding a typecheck error (e.g. when running `brew tests --only=dev-cmd/livecheck --online`).